### DISCUSSION TITLE
is "page" from "answers_page_bounding_boxes" reliable?

### DISCUSSION THREAD

**Discussion by jianglong-he-Infrrd** [duchallenge-team/dude#8] [Q&A] [answered]
Hi,

For extractive questions, how reliable are the `"page" `information? I have seen wrong answer page numbers. For ex: questionId `"c71ce4a92fd6ac98abb1b7e1e9332f9b_a0d9e0cddb840986f97c5673bb114eec"`

I also need clarification on following points:
1. what does `"error": "Box draw over grid padding"` mean from `"answers_page_bounding_boxes"`?
2. is answer page number `"page"` using zero page indexing?

Thank you!

**Comment by pjoziak** [comment]
Hello @jianglong-he-Infrrd,

thanks for you interest in our challenge. 

tl;dr
1. Annotation artifact, not yielding any crucial information.
2. Yes, 0-based indexing.

Long answer

In order to answer your questions, let me please explain something.
Many documents went through several rounds of annotation (up to 4), and the `answers_page_bounding_boxes` field is a result of a heuristic of ours that represents how we found an agreement between bounding boxes selected by different annotators. We paid attention mostly to whether the rectangles differ, and if so -- by how much, and depending of the result, we either sent the document for yet another round of annotation, or we have taken the intersection/span of the rectangles, depending on certain characteristics.

Regarding your questions:
1. it is an error produced the annotation widget, and it essentially means that in one of the phases the annotator performed a misclick and tried to select a bounding box outside of an active area. When joining bounding box annotation from different rounds of annotations, these error logs leaked into the final annotation json file. Nevertheless, the bounding box coordinates should be -- to the best of our knowledge -- properly expressed, as this is a result of the intersection/span operation.
2. it's 0-based indexing, 1 off to the artificial page number generated on top of the original document image, in the bottom-right corner of each page. So a part of the picture labelled "6" will be represented as `"page": 5` in the ground truth JSON.

To comment the error you mentioned, I've checked that there are precisely 208 bounding box annotations out of 14777 in total, which might be dubious, some 1.4%. However, even in case of those the rectangle itself should be fine. 
What you can do to make sure whether a bounding box annotation is right, and whether or not to include it in your data model, is:
- check that the OCR bounding boxes which fit/significantly overlap with the proposed  answer bounding box contain strings similar to the ground truth answer (note the answers might have a different form, so it's not necessarily string equality, e.g. numbers, dates, currencies could have been normalized into standard form[^1])
- recalculate the page number based on bounding box coordinates. This might be of use for those 208 dubious annotations.


[^1]: the standard form representation is taken into account in official evaluation script, so there is no need to make sure your model produces a normalized output. E.g. the document can speak of `November 5th, 2006`, but the ground truth annotation for it is `05-11-2006`. If your model outputs any representation of this date: `November 5th, 2006` or `05-11-2006` or even `2006-11-05` or `5/11/06` we will treat it as the correct answer)


In case any further support is needed, we'll do our best to help.

**Reply by jianglong-he-Infrrd** [reply]
Hi @pjoziak ,

Thanks for the clarification. I just have one more doubt, are we sure its 1-based indexing? I have seen a lot `"page": 0` in `"answers_page_bounding_boxes"` field.

**Comment by pjoziak** [comment, marked as answer]
You are right, it is 0-based indexing. 
Previously I was looking at a "raw" version of the dataset, sorry for the confusion I caused.
